modules/wallet/files_test.go

```diff
@@ -0,0 +1,36 @@
+package wallet
+
+import (
+	"testing"
+)
+
+// TestSaveLoad tests that saving and loading a wallet restores its data.
+func TestSaveLoad(t *testing.T) {
+	wt := NewWalletTester(t)
+	// add an output to the wallet
+	wt.testCoinAddress()
+
+	// save wallet data
+	err := wt.save()
+	if err != nil {
+		wt.Fatal(err)
+	}
+
+	// create a new wallet using the saved data
+	newWallet, err := New(wt.state, wt.tpool, wt.filename)
+	if err != nil {
+		wt.Fatal(err)
+	}
+
+	// check that the wallets match
+	for mapKey := range wt.keys {
+		if _, exists := newWallet.keys[mapKey]; !exists {
+			wt.Fatal("Loaded wallet is missing a key")
+		}
+	}
+	for mapKey := range wt.timelockedKeys {
+		if _, exists := newWallet.timelockedKeys[mapKey]; !exists {
+			wt.Fatal("Loaded wallet is missing a time-locked key")
+		}
+	}
```

Comment:
Can you add some code to make sure the balances match? I doubt the test will fail but I still think the test should be in place.